### PR TITLE
Updated BackEndApplication.py in iot-hub/Quickstarts for v2 SDK

### PR DESCRIPTION
**Pull request by JimacoMS3**
## Purpose

* Update BackendApplication.py to use V2 SDK

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x ] Other... Please describe: Updating sample for new version of SDK
```

## How to Test
*  Get the code

```
git clone https://github.com/JimacoMS3/azure-iot-samples-python.git
cd azure-iot-samples-python
git checkout v2-backend-change
```

* Test the code

1. Set up IoT hub with a device named MyPythonDevice.
2. Get IoT Hub service policy connection string and device connection string
3. Run Device
    a. Open a command prompt and cd to azure-iot-samples-python/iot-hub/Quickstarts/simulated-device
    b. In an editor change the connection string in SimulatedDevice.py to your device connection string
   c. In command prompt do `pip install azure-iot-device` followed by `python SimulatedDevice.py`
   d. You should see telemetry being sent every second or so.

4. Run Backend
    a. Open another command prompt and cd to azure-iot-samples-python/iot-hub/Quickstarts/back-end-application
    b. In an editor change the connection string in BackEndApplication.py to your IoT Hub service connection string
   c. In command prompt do `pip install azure-iot-hub` followed by `python BackEndApplication.py`
   d. You should see the direct method invocation and response. In the device command window you should see the telemetry rate slow to every 10  seconds or so.

For more detail on steps see the following PR: https://github.com/MicrosoftDocs/azure-docs-pr/pull/100467 and preview the topic.

 
```

## What to Check
Verify that the following are valid
* See above. you should see the command being sent and a response and the telemetry slow to every 10 seconds.

## Other Information
<!-- Add any other helpful information that may be needed here. -->

### CODE REVIEW

**Comment by JimacoMS3** [comment]
@cartertinney and @jebrando please review this PR. Please note that it is related to the following PR in docs: https://github.com/MicrosoftDocs/azure-docs-pr/pull/100467, on which your review is also requested.

File: iot-hub/Quickstarts/back-end-application/BackEndApplication.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
-import sys
-
 # Copyright (c) Microsoft. All rights reserved.
 # Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 # Using the Python Device SDK for IoT Hub:
 #   https://github.com/Azure/azure-iot-sdk-python
-# The sample connects to a device-specific MQTT endpoint on your IoT Hub.
-import iothub_service_client
+# The sample connects to a device-specific HTTP endpoint on your IoT Hub.
```
Comment (JimacoMS3) [inline iot-hub/Quickstarts/back-end-application/BackEndApplication.py]: @cartertinney @jebrando Please validate this statement about HTTP

File: iot-hub/Quickstarts/back-end-application/BackEndApplication.py [hunk 2]
```diff
@@ -40,9 +41,9 @@ def iothub_devicemethod_sample_run():
 
         input("Press Enter to continue...\n")
 
-    except IoTHubError as iothub_error:
+    except Exception as ex:
```
Comment (JimacoMS3) [inline iot-hub/Quickstarts/back-end-application/BackEndApplication.py]: @jebrando @cartertinney I couldn't figure out how to get hub specific errors (as we did previously). LMK if this needs to be more specific.

Comment (jebrando) [reply iot-hub/Quickstarts/back-end-application/BackEndApplication.py]: For now the ex is the most detail we have for errors.

---
In reply to: [364444185](https://github.com/Azure-Samples/azure-iot-samples-python/pull/20#discussion_r364444185) [](ancestors = 364444185)

**Comment by JimacoMS3** [comment]
#sign-off

**Pull request merged**